corehq/apps/custom_data_fields/migrations/0009_field_required_for.py

```diff
@@ -0,0 +1,35 @@
+# Generated by Django 4.2.15 on 2024-10-09 17:52
+
+from django.db import migrations, models
+
+from corehq.util.django_migrations import skip_on_fresh_install
+
+CUSTOM_USER_DATA_FIELD_TYPE = 'UserFields'
+DEFAULT_REQUIRED_FOR = ['commcare_user']
+
+
+@skip_on_fresh_install
+def update_user_fields_required_for(apps, schema_editor):
+    Field = apps.get_model('custom_data_fields', 'Field')
+    CustomDataFieldsDefinition = apps.get_model('custom_data_fields', 'CustomDataFieldsDefinition')
+
+    user_fields_definitions = CustomDataFieldsDefinition.objects.filter(field_type=CUSTOM_USER_DATA_FIELD_TYPE)
+
+    for definition in user_fields_definitions:
+        Field.objects.filter(definition=definition).update(required_for=DEFAULT_REQUIRED_FOR)
```

Comment:
Isn't this a change from current behavior? From skimming the ticket and design document, it sounds like right now fields are required for both mobile and web users. It sounds like mobile workers only is the desired default behavior, but are you certain this change won't affect existing projects in a way they don't expect?

Reply:
Yes it is a change in current behavior and you're correct that right now, fields are required for both mobile and web users. We discussed/are discussing it in [this](https://docs.google.com/document/d/1QiTdoDQPPT5x2fEie6bvZD1qsUVhtSlaTTSzARW9jPg/edit?disco=AAABVQJrlRA) comment thread. The decision from that discussion thread was to migrate existing projects to have required fields apply to "mobile worker".

But you're right that it would be disruptive to change current behavior. So I think the better strategy would be to migrate such that required fields still apply to both web users and mobile workers. And then domains who do not want required fields applied to web users can configure it as such.

Reply:
Changed migration such that existing required fields continue to apply to both web users and mobile workers [fa7b8d4](https://github.com/dimagi/commcare-hq/pull/35218/commits/fa7b8d43e4223ccf382bb9803e4b707f127bea4d) [9a56a14](https://github.com/dimagi/commcare-hq/pull/35218/commits/9a56a14873ee94acb08f4a31cb932eb2fdf50c0f) Thank you for flagging this, definitely would have caused commotion if I had migrated existing fields to apply only for mobile workers.